src/elexclarity/formatters.py

```diff
@@ -18,16 +18,17 @@ def get_id_for_choice(self, name):
         """
         return slugify(name)
 
-    def get_id_for_precinct(self, fips, name):
+    def get_subunit_id(self, subunit_name, fips=None):
         """
-        Create an ID from the clarity key and candidate name. Unclear
-        if these precinct names are unique across counties (assume
-        they are not), so prepent the county fips.
+        Create an ID for a precinct or county.
+        Fips codes are present when level == precinct.
         """
-        name = slugify(name)
+        name = slugify(subunit_name)
+        if fips is None:
+            fips = self.county_lookup.get(subunit_name)
```

Comment:
I am probably misunderstanding, but tracing this back it seems like we've already done the FIPS lookup but are doing it again here when level == county?

Reply:
Yep, the way Anthony's PR had this set up was when you're handling county-level XML data to parse precincts, you pass down a county fips code to all of its affiliated precincts because you can't access the county name in the <Precinct/> tag, but when handling state-level XML, you access the county names directly in the <County/> tag